### PR TITLE
Features/backend/comments

### PR DESCRIPTION
**Pull request by Jeaksen**
Comments functionality implemented:
- /lostdogs/{dogId}/comments POST - add a comment to a dog
- /lostdogs/{dogId}/comments/{commentId} DELETE - remove a comment (as comment author)

### CODE REVIEW

**Review by MateuszKubiszewski — APPROVED** [review]

**Review by emiwroblewska — APPROVED** [review]

**Review by PjuralSzymon — APPROVED** [review]

**Review by MateuszZalewski0311 — COMMENTED** [review]
Typos in function names in LostDogServiceTests and LostDogController

File: Backend/Backend.Tests/LostDogs/LostDogServiceTests.cs [hunk 1]
```diff
@@ -185,12 +185,119 @@ public async void UpdatingLostDogFailsForNonExistingDogNewPicture()
...
+            var security = new Mock<ISecurityService>();
+
+            repo.Setup(o => o.DeleteLostDogComment(It.IsAny<int>(), It.IsAny<int>())).Returns(Task.FromResult(new RepositoryResponse()));
+            var service = new LostDogService(repo.Object, security.Object, mapper, logger);
+
+            Assert.True((await service.DeleteLostDogComment(1, 1)).Successful);
+        }
+
+        [Fact]
+        public async void DeleteLostDogCommentSuccessfulForRepoError()
```
Comment (MateuszZalewski0311) [inline Backend/Backend.Tests/LostDogs/LostDogServiceTests.cs:268]: Function Name Typo

File: Backend/Backend.Tests/LostDogs/LostDogServiceTests.cs [hunk 2]
```diff
@@ -185,12 +185,119 @@ public async void UpdatingLostDogFailsForNonExistingDogNewPicture()
...
+            var security = new Mock<ISecurityService>();
+
+            repo.Setup(o => o.GetLostDogComment(It.IsAny<int>(), It.IsAny<int>())).Returns(Task.FromResult(new RepositoryResponse<LostDogComment>()));
+            var service = new LostDogService(repo.Object, security.Object, mapper, logger);
+
+            Assert.True((await service.GetLostDogComment(1, 1)).Successful);
+        }
+
+        [Fact]
+        public async void GetLostDogCommentSuccessfulForRepoError()
```
Comment (MateuszZalewski0311) [inline Backend/Backend.Tests/LostDogs/LostDogServiceTests.cs:292]: Function Name Typo

File: Backend/Backend/Controllers/LostDogController.cs [hunk 3]
```diff
@@ -143,22 +143,42 @@ public async Task<IActionResult> DeleteLostDog(int dogId)
...
+            comment.DogId = dogId;
+
+            var serviceResponse = await lostDogService.AddLostDogComment(comment, picture);
+            var controllerResponse = mapper.Map<ControllerResponse<GetCommentDto>>(serviceResponse);
+            return StatusCode(serviceResponse.StatusCode, controllerResponse);
+        }
+
+        [HttpDelete]
+        [Route("{dogId}/comments/{commentId}")]
+        public async Task<IActionResult> AddLostDogComment([FromRoute] int dogId, [FromRoute] int commentId)
```
Comment (MateuszZalewski0311) [inline Backend/Backend/Controllers/LostDogController.cs:162]: Function Name Typo

**Pull request merged**